packages/builder/src/pages/builder/app/[application]/design/[assetType]/_layout.svelte

```diff
@@ -16,13 +18,25 @@
   import AppThemeSelect from "components/design/AppPreview/AppThemeSelect.svelte"
   import ThemeEditor from "components/design/AppPreview/ThemeEditor.svelte"
   import DevicePreviewSelect from "components/design/AppPreview/DevicePreviewSelect.svelte"
+  import NavigationSelectionModal from "components/design/NavigationPanel/NavigationSelectionModal.svelte"
+  import ScreenNameModal from "components/design/NavigationPanel/ScreenNameModal.svelte"
+  import NewScreenModal from "components/design/NavigationPanel/NewScreenModal.svelte"
+  import Logo from "assets/bb-space-man.svg"
 
   // Cache previous values so we don't update the URL more than necessary
   let previousType
   let previousAsset
   let previousComponentId
   let hydrationComplete = false
 
+  // Manage the layout modal flow from here
+  let modal
```

Comment:
It might be a good idea at this stage to create a new file that's in charge of the screen creation process, and remove all these modals from the `_layout.svelte` file. That file can be in control of showing and hiding modals and it can contain the various state properties that are currently defined here. Basically just cut and paste all this screen creation logic and modals to a single file.

Since there are now multiple modals involved in the screen creation process, I don't think it's very good practice to pass around references to all the modals to each of them and let them be in control of showing and hiding each other. It makes maintaining it very difficult since there's no obivous way to see the control flow between them, and also couples them together strongly.

If it's too much work to refactor to remove the modal references being passed in to each other then that's alright I guess, but I think it would be a much cleaner solution if the modals themselves didn't have to be aware of each other, and the controlling file was the one in charge, and could show and hide the correct modal and use callbacks from the other modals to keep track of state.

Reply:
updated this to be controlled from ScreenWizard.svelte, it's much cleaner.